pkg/api/v1alpha1/cloudstackmachineconfig_webhook.go

```diff
@@ -58,7 +60,7 @@ func (r *CloudStackMachineConfig) ValidateUpdate(old runtime.Object) error {
 		return nil
 	}
 
-	if oldCloudStackMachineConfig.IsManagement() {
+	if oldCloudStackMachineConfig.IsManagement() && !features.IsActive(features.ExperimentalSelfManagedClusterUpgrade()) {
```

Comment:
I'm confused by this statement
should it be not negated?

Reply:
If the cluster is a management cluster, it used to return from this line without verification. Thus, I have added one more check if the feature is active it should allow to verify further and not return from here.